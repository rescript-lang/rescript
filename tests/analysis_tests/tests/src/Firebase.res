module Firebase = {
  module Firestore = {
    type firestore

    type documentReference<'documentdata> = {
      id: string,
      path: string,
    }

    type documentSnapshot<'documentdata> = {
      id: string,
      ref: documentReference<'documentdata>,
    }

    @module("firebase/firestore") @variadic
    external doc: (firestore, string, array<string>) => documentReference<'documentdata> = "doc"

    @module("firebase/firestore")
    external getDoc: documentReference<'documentdata> => Promise.t<
      documentSnapshot<'documentdata>,
    > = "getDoc"
  }
}

module Sample = {
  open Firebase

  external store: Firestore.firestore = "store"

  let ref = store->Firestore.doc("some_id", [])
  // ref.
  //     ^com
}
