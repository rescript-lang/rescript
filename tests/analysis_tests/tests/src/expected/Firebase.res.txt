Complete src/Firebase.res 30:9
posCursor:[30:9] posNoWhite:[30:8] Found expr:[30:5->30:9]
Pexp_field [30:5->30:8] _:[32:0->30:9]
Completable: Cpath Value[ref].""
Raw opens: 1 Firebase.place holder
Package opens Stdlib.place holder Pervasives.JsxModules.place holder
Resolved opens 2 Stdlib Firebase
ContextPath Value[ref].""
ContextPath Value[ref]
Path ref
ContextPath Value[ref]->
ContextPath Value[ref]
Path ref
CPPipe pathFromEnv:Firebase.Firestore found:true
Path Firebase.Firestore.
[{
    "label": "id",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\nid: string\n```\n\n```rescript\ntype documentReference<'documentdata> = {\n  id: string,\n  path: string,\n}\n```"}
  }, {
    "label": "path",
    "kind": 5,
    "tags": [],
    "detail": "string",
    "documentation": {"kind": "markdown", "value": "```rescript\npath: string\n```\n\n```rescript\ntype documentReference<'documentdata> = {\n  id: string,\n  path: string,\n}\n```"}
  }, {
    "label": "->Firestore.getDoc",
    "kind": 12,
    "tags": [],
    "detail": "documentReference<\n  'documentdata,\n> => Stdlib.Promise.t<documentSnapshot<'documentdata>>",
    "documentation": null,
    "sortText": "getDoc",
    "insertText": "->Firestore.getDoc",
    "additionalTextEdits": [{
      "range": {"start": {"line": 30, "character": 8}, "end": {"line": 30, "character": 9}},
      "newText": ""
      }]
  }]

